doc/loader.rst

```diff
@@ -5,128 +5,539 @@ MIG LOADER
+
+This package will contain the mig-loader binary built with our configuration, which contains the
+API URL the loader should use and the GPG keys that will be used to validate incoming manifests. Next
+the package can be installed on the system we want to run the agent on.
+
+Loader configuration and initial run
+------------------------------------
+The loader should be setup to run periodically on the system. This ensures the device periodically
+checks for updates, and installs new agent code when required. The periodic job configuration depends
+on the operating system the loader is installed on. For Linux based devices, typically ``mig-loader``
+would be setup to run as root via a cron entry, or if cron is not on the system using a systemd
+timer. On Darwin, the installer automatically creates an interval based launchd job to run the loader.
+
+Put the loader key for this instance into ``/etc/mig/mig-loader.key``. This should contain the key
+we used to create the loader instance on a single line.
+
+We can run ``/sbin/mig-loader`` manually on the system now.
+
+::
+
+    # /sbin/mig-loader
+    logging routine started
+    Ident is Ubuntu 15.10 wily
+    Init is upstart
+    leaving findOSInfo()
+    Found local address 10.0.0.18/24
+    Found public ip 10.0.0.18
+    AWS metadata service not found, skipping fetch
+    initialized local bundle information
+    mig-agent /sbin/mig-agent -> not found
+    mig-loader /sbin/mig-loader -> 40d83204825421c82379b65b8c7077fd110a4af5391acfc8052e568d0830af26
+    configuration /etc/mig/mig-agent.cfg -> not found
+    agentcert /etc/mig/agent.crt -> not found
+    agentkey /etc/mig/agent.key -> not found
+    cacert /etc/mig/ca.crt -> not found
+    requesting manifest from https://my.mig.api.url:1664/api/v1/manifest/agent/
+    1 valid signatures in manifest
+    comparing mig-agent /sbin/mig-agent
+    we have not found
+    they have d3bc2fdbd42404f2df9472d8de900889f8755d12041cda7f65fa7ba99e3eeda3
+    refreshing mig-agent
+    fetching file from https://my.mig.api.url:1664/api/v1/manifest/fetch/
+    validating staged file signature
+    renaming existing file
+    installing staged file
+    comparing mig-loader /sbin/mig-loader
+    we have 40d83204825421c82379b65b8c7077fd110a4af5391acfc8052e568d0830af26
+    they have 3d584ad090c556234ad6148006ab0dcd693ab9f99c386413a8597034420384dc
+    refreshing mig-loader
+    fetching file from https://my.mig.api.url:1664/api/v1/manifest/fetch/
+    validating staged file signature
+    renaming existing file
+    installing staged file
+    comparing configuration /etc/mig/mig-agent.cfg
+    we have not found
+    they have d51a2e9d955aaca94e88159ad6235cbaccf9680f0d8e82dcee0f2f0f0df83038
+    refreshing configuration
+    fetching file from https://my.mig.api.url:1664/api/v1/manifest/fetch/
+    validating staged file signature
+    renaming existing file
+    installing staged file
+    comparing agentcert /etc/mig/agent.crt
+    we have not found
```

Comment:
`we have not found` ....
proposal:
* your underwear
* the cat
* any beer in the fridge
* the answer to life